wifiphisher/common/interfaces.py

```diff
@@ -217,180 +272,220 @@ def __init__(self):
-
-    def get_jam_iface(self, interface_name):
-        for k, interface in self._interfaces.iteritems():
-            if k == interface_name and not interface.being_used:
-                if interface.has_monitor_mode():
-                    return interface
-                else:
-                    raise JammingInterfaceInvalidError
-        raise JammingInterfaceInvalidError
-
-    def get_ap_iface(self, interface_name=None):
-        for k, interface in self._interfaces.iteritems():
-            if interface_name == None:
-                if interface.has_ap_mode():
-                    return interface
-            if k == interface_name and not interface.being_used:
-                if interface.has_ap_mode():
-                    return interface
+        """
+
+        monitor_interface = self.get_interface(has_monitor_mode=True)
+        ap_interface = self.get_interface(has_ap_mode=True)
+
+        return (monitor_interface, ap_interface)
+
+    def is_interface_wired(self, interface_name):
+        """
+        Check whether the interface is wired or not
+
+        :param self: A NetworkManager object
+        :param interface_name: Name of an interface
+        :type self: NetworkManager
+        :type interface_name: string
+        :return: None
+        :rtype: None
+        :raises InvalidInternetInterfaceError: If interface is not wired
+        """
+
+        if self._name_to_object[interface_name].is_wireless:
+            raise InvalidInternetInterfaceError(interface_name)
+
+    def unblock_interface(self, interface_name):
+        """
+        Unblock interface if it is blocked
+
+        :param self: A NetworkManager object
+        :param interface_name: Name of an interface
+        :type self: NetworkManager
+        :type interface_name: string
+        :return: None
+        :rtype: None
+        """
+
+        card = self._name_to_object[interface_name].card
+
+        # unblock card if it is blocked
+        if pyw.isblocked(card):
+            pyw.unblock(card)
+
+    def set_interface_channel(self, interface_name, channel):
+        """
+        Set the channel for the interface
+
+        :param self: A NetworkManager object
+        :param interface_name: Name of an interface
+        :param channel: A channel number
+        :type self: NetworkManager
+        :type interface_name: string
+        :type channel: int
+        :return: None
+        :rtype: None
+        """
+
+        card = self._name_to_object[interface_name].card
+
+        pyw.chset(card, channel)
+
+    def start(self):
+        """
+        Start the network manager
+
+        :param self: A NetworkManager object
+        :type self: NetworkManager
+        :return: None
+        :rtype: None
+        """
+
+        # populate our dictionary with all the available interfaces on the system
+        for interface in pyw.interfaces():
+            try:
+                card = pyw.getcard(interface)
```

Comment:
This seems to throw a `pyric.error` (Errno 19: No such device)

The problem is that that `pyw.interfaces` returns all interfaces in the system including loopback but there's no device card tied to the loopback interface.